File: 20240802/H-index.cpp
#include <vector>
#include <algorithm>
using namespace std;

int answer;

int solution(vector<int> citations) {
    sort(citations.begin(), citations.end(), greater<int>());
    
    // i번째 논문이 (i+1)번 이상 인용되지 않을 때까지 정답을 하나씩 증가
    for(int i=0; i<citations.size(); ++i) {
        if(citations[i] < i + 1) break;
        ++answer;
    }
    
    return answer;
}

// https://resilient-aura-65d.notion.site/H-index-378c8d8762d94f91836ec96c07cdf8ec?pvs=4

File: 20240802/HIndex.java
import java.util.*;
class HIndex {
    public int hIndex(int[] citations) {
        int answer = 0;
        Arrays.sort(citations);
        //h이상의 수기 때문에 배열 전체 길이와 배열에서 h수 뺀 곳의 값이 같으면 h-index
        for(int i=0; i<citations.length; i++){
            int h = citations.length-i;
            if(citations[i]>=h){
                answer = h;
                break;
            }
        }
        return answer;
    }
}
//18:28 ~ 18:41